File: GdPicture/WinForm/C#/Adjusting Gamma - Brightness - Contrast & Saturation/Form1.cs
using GdPicture14;
using System;
using System.Windows.Forms;

namespace adjust
{
    internal partial class Form1 : Form
    {
        private int _originalImageId;

        private readonly GdPictureImaging _oGdPictureImaging = new GdPictureImaging();


        private void Form1_Load(System.Object eventSender, System.EventArgs eventArgs)
        {
            LicenseManager oLicenseManager = new LicenseManager(); //Go to http://www.gdpicture.com/download-gdpicture/ to get a 1 month trial key unlocking all features of the toolkit.
            oLicenseManager.RegisterKEY("XXXX"); //Please, replace XXXX by a valid demo or commercial license key.
        }


        private void StripMenuItemOpenTool_Click(System.Object sender, System.EventArgs e)
        {
            ReleaseGdPictureImage();
            ResetBars();

            _originalImageId = _oGdPictureImaging.CreateGdPictureImageFromFile(""); //Providing an empty string prompts the user to select a file.
            if (_originalImageId != 0)
            {
                GdPictureStatus gdPictureStatus;
                if ((gdPictureStatus = GdVieware1.DisplayFromGdPictureImage(_originalImageId)) != GdPictureStatus.OK)
                {
                    MessageBox.Show(gdPictureStatus.ToString(), "Image could not be displayed", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            else
            {
                MessageBox.Show(_oGdPictureImaging.GetStat().ToString(), "Image could not be loaded", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }


        private void Apply()
        {
            if (_originalImageId != 0)
            {
                //Cloning the image to perform modifications on a copy so that the original one remains unaltered.
                int clonedImageId = _oGdPictureImaging.CreateClonedGdPictureImage(_originalImageId);
                if (clonedImageId != 0)
                {
                    //Changing the value of the calling scrollbar accordingly to the scroll of the user.
                    //Adjusting the brightness, the contrast, the saturation and the gamma of the image.
                    GdPictureStatus gdPictureStatus = _oGdPictureImaging.SetBCSG(clonedImageId,
                        HScrollBarBrightness.Value, HScrollBarContrast.Value,
                        HScrollBarGamma.Value, HScrollBarSaturation.Value);

                    if (gdPictureStatus == GdPictureStatus.OK)
                    {
                        if ((gdPictureStatus = GdVieware1.DisplayFromGdPictureImage(clonedImageId)) != GdPictureStatus.OK)
                        {
                            MessageBox.Show(gdPictureStatus.ToString(), "Image could not be displayed", MessageBoxButtons.OK, MessageBoxIcon.Error);
                        }
                    }
                    else
                    {
                        MessageBox.Show(gdPictureStatus.ToString(), "BCSG could not be adjusted", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }

                    //Deleting the cloned image from memory.
                    _oGdPictureImaging.ReleaseGdPictureImage(clonedImageId);
                }
                else
                {
                    MessageBox.Show(_oGdPictureImaging.GetStat().ToString(), "Image could not be cloned", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
        }


        private void ScrollBarBrightness_Scroll(System.Object eventSender, ScrollEventArgs eventArgs)
        {
            if (eventArgs.OldValue != eventArgs.NewValue)
            {
                Apply();
            }
        }


        private void ScrollBarContrast_Scroll(System.Object eventSender, ScrollEventArgs eventArgs)
        {
            if (eventArgs.OldValue != eventArgs.NewValue)
            {
                Apply();
            }
        }


        private void ScrollBarGamma_Scroll(System.Object eventSender, ScrollEventArgs eventArgs)
        {
            if (eventArgs.OldValue != eventArgs.NewValue)
            {
                Apply();
            }
        }


        private void ScrollBarSaturation_Scroll(System.Object eventSender, ScrollEventArgs eventArgs)
        {
            if (eventArgs.OldValue != eventArgs.NewValue)
            {
                Apply();
            }
        }


        private void ResetBars()
        {
            HScrollBarSaturation.Value = 0;
            HScrollBarGamma.Value = 0;
            HScrollBarContrast.Value = 0;
            HScrollBarBrightness.Value = 0;
        }


        private void Reset_Click(object sender, EventArgs e)
        {
            GdPictureStatus gdPictureStatus;
            if ((gdPictureStatus = GdVieware1.DisplayFromGdPictureImage(_originalImageId)) != GdPictureStatus.OK)
            {
                System.Windows.Forms.MessageBox.Show(gdPictureStatus.ToString(), "Image could not be displayed", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            ResetBars();
        }


        private void ReleaseGdPictureImage()
        {
            if (_originalImageId != 0)
            {
                //Deleting the image from memory.
                _oGdPictureImaging.ReleaseGdPictureImage(_originalImageId);
                _originalImageId = 0;
            }
            //Clearing the viewer.
            GdVieware1.CloseDocument();
        }


        private void CloseToolStripMenuItem_Click(System.Object sender, System.EventArgs e)
        {
            ReleaseGdPictureImage();
            ResetBars();
        }


        private void ExitToolStripMenuItem_Click(System.Object sender, System.EventArgs e)
        {
            ReleaseGdPictureImage();
            Close();
        }
    }
}

File: GdPicture/WinForm/vb.net/Adjusting Gamma - Brightness - Contrast & Saturation/Form1.vb
Option Strict On
Option Explicit On

Imports GdPicture14


Public Class Form1

    Private _sourceImage As Integer
    Private _clonedImage As Integer
    Private _gdPictureStatus As GdPictureStatus
    Private ReadOnly _gdPictureImaging As New GdPictureImaging



    Private Sub Form1_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
        Dim oLicenseManager As New GdPicture14.LicenseManager() 'Go to http://www.gdpicture.com/download-gdpicture/ to get a 1 month trial key unlocking all features of the toolkit.
        oLicenseManager.RegisterKEY("XXXX") 'Please, replace XXXX by a valid demo or commercial license key.
    End Sub

    Private Sub HScroll1_Scroll(ByVal eventSender As System.Object, ByVal eventArgs As ScrollEventArgs) Handles HScroll1.Scroll
        If eventArgs.OldValue <> eventArgs.NewValue Then
            ProcessFilters()
        End If
    End Sub

    Private Sub HScroll2_Scroll(ByVal eventSender As System.Object, ByVal eventArgs As ScrollEventArgs) Handles HScroll2.Scroll
        If eventArgs.OldValue <> eventArgs.NewValue Then
            ProcessFilters()
        End If
    End Sub

    Private Sub HScroll3_Scroll(ByVal eventSender As System.Object, ByVal eventArgs As ScrollEventArgs) Handles HScroll3.Scroll
        If eventArgs.OldValue <> eventArgs.NewValue Then
            ProcessFilters()
        End If
    End Sub

    Private Sub HScroll4_Scroll(ByVal eventSender As System.Object, ByVal eventArgs As ScrollEventArgs) Handles HScroll4.Scroll
        If eventArgs.OldValue <> eventArgs.NewValue Then
            ProcessFilters()
        End If
    End Sub

    Private Sub ProcessFilters()
        If _sourceImage <> 0 Then
            If _clonedImage <> 0 Then
                _gdPictureImaging.ReleaseGdPictureImage(_clonedImage)
                _clonedImage = 0
            End If
            _clonedImage = _gdPictureImaging.CreateClonedGdPictureImage32BppPARGB(_sourceImage)

            _gdPictureStatus = _gdPictureImaging.SetBCSG(_clonedImage, HScroll1.Value, HScroll2.Value, HScroll4.Value, HScroll3.Value)

            If _gdPictureStatus = 0 Then
                GdViewer1.DisplayFromGdPictureImage(_clonedImage)
            Else
                MsgBox("Error No: " & _gdPictureStatus.ToString)
            End If
        End If
    End Sub


    Private Sub OpenToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OpenToolStripMenuItem.Click
        Call ReleaseGdPictureImage()

        _sourceImage = _gdPictureImaging.CreateGdPictureImageFromFile("")
        If _gdPictureImaging.GetStat = 0 Then

            If (_gdPictureImaging.GetWidth(_sourceImage) * _gdPictureImaging.GetHeight(_sourceImage)) > (GdViewer1.Width * GdViewer1.Height) Then
                'Resizing the image to viewer height to maximize performances
                _gdPictureImaging.ResizeWidthRatio(_sourceImage, GdViewer1.Width, Drawing2D.InterpolationMode.HighQualityBicubic)
            End If
            ProcessFilters()
        End If
    End Sub

    Private Sub ReleaseGdPictureImage()
        If _sourceImage <> 0 Then
            _gdPictureImaging.ReleaseGdPictureImage(_sourceImage)
            _sourceImage = 0
        End If
        If _clonedImage <> 0 Then
            _gdPictureImaging.ReleaseGdPictureImage(_clonedImage)
            _clonedImage = 0
        End If
        GdViewer1.CloseDocument()
    End Sub

    Private Sub CloseToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CloseToolStripMenuItem.Click
        ReleaseGdPictureImage()
    End Sub

    Private Sub ExitToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ExitToolStripMenuItem.Click
        ReleaseGdPictureImage()
        Me.Close()
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        HScroll1.Value = 0
        HScroll2.Value = 0
        HScroll3.Value = 0
        HScroll4.Value = 0
        ProcessFilters()
    End Sub
End Class